extensions/argo/numa-rollout/numa-rollout-extension/ui/src/rollout/default/ArgoRolloutComponent.tsx

```diff
@@ -0,0 +1,313 @@
+  }
+  return "0";
+};
+
+const parseRevision = (node: any) => {
+  for (const item of node.info || []) {
+    if (item.name === "Revision") {
+      const parts = item.value.split(":") || [];
+      return parts.length === 2 ? parts[1] : "0";
+    }
+  }
+};
+
+const parsePodStatus = (pod: any) => {
+  for (const item of pod.info || []) {
+    if (item.name === "Status Reason") {
+      return item.value;
+    }
+  }
+};
+
+const parseAnalysisRuns = (
+  app: any,
+  tree: any,
+  rollout: any
+): RolloutAnalysisRunInfo[] => {
+  const [analysisRunResults, setAnalysisRunResults] = React.useState<
+    RolloutAnalysisRunInfo[]
+  >([]);
+  const [analysisRunNodeIds, setAnalysisRunNodeIds] = React.useState<string[]>(
+    []
+  );
+  const [isRefresh, setIsRefresh] = React.useState<boolean>(false);
+
+  // Get the list of AnalysisRun node IDs whenever the tree or rollout props change
+  React.useMemo(() => {
+    const filteredNodes = tree.nodes.filter(
+      (node: any) =>
+        node.kind === "AnalysisRun" &&
+        node.parentRefs.some((ref: any) => ref.name === rollout.metadata.name)
+    );
+    const nodeIds = filteredNodes.map((node: any) => node.uid);
+
+    // Check if there are any new AnalysisRun node IDs or if the count has changed from previous node IDs
+    if (
+      nodeIds.length !== analysisRunNodeIds.length ||
+      !analysisRunNodeIds.every((value, index) => value === nodeIds[index])
+    ) {
+      setIsRefresh(true);
+    }
+    setAnalysisRunNodeIds(nodeIds);
+  }, [tree.nodes]);
+
+  const rolloutAnalysisRunInfo = async () => {
+    const promises: Promise<RolloutAnalysisRunInfo>[] = analysisRunNodeIds.map(
+      async (nodeId) => {
+        const node: any = tree.nodes.find((node: any) => node.uid === nodeId);
+
+        const state = await getResource(
+          app.metadata.name,
+          app.metadata.namespace,
+          node as any
+        );
+        return {
+          objectMeta: {
+            creationTimestamp: {
+              seconds: node.createdAt,
+            },
+            name: node.name,
+            namespace: node.namespace,
+            resourceVersion: node.version,
+            uid: node.uid,
+          },
+          specAndStatus: {
+            spec: state.spec,
+            status: state.status || null,
+          },
+          revision: parseRevision(node),
+          status: parseAnalysisRunStatus(node.health.status),
+        };
+      }
+    );
+
+    const newAnalysisRunResults = await Promise.all(promises);
+    setIsRefresh(false);
+    setAnalysisRunResults(newAnalysisRunResults);
+  };
+  // Call the API call function only when isRefresh is true and AnalysisRun node IDs exist
+  React.useEffect(() => {
+    if (isRefresh && analysisRunNodeIds.length > 0) {
+      rolloutAnalysisRunInfo();
+    }
+  }, [isRefresh, analysisRunNodeIds]);
+
+  return analysisRunResults;
+};
+
+const parseAnalysisRunStatus = (status: string): string => {
+  switch (status) {
+    case "Healthy":
```

Comment:
can we move all string to variables? like `const HEALTHY = "Healthy"

Reply:
Done!